### PR TITLE
Corrected Polygamma function and tests.

### LINKED ISSUE #806: Polygamma(0, 1/4) is wrong
`add(neg(div(pi, i3)), sub(mul(im3, log(i2)), EulerGamma))` which is `-pi/3 -3*log(2) - EulerGamma` but needs to be `-3*log(2) - pi/2 - EulerGamma`

See sympy algorithm,

```Python
lookup = {S(1)/2: -2*log(2) - S.EulerGamma,
          S(1)/3: -S.Pi/2/sqrt(3) - 3*log(3)/2 - S.EulerGamma,
          S(1)/4: -S.Pi/2 - 3*log(2) - S.EulerGamma,
          S(3)/4: -3*log(2) - S.EulerGamma + S.Pi/2,
          S(2)/3: -3*log(3)/2 + S.Pi/2/sqrt(3) - S.EulerGamma}
if z > 0:
    n = floor(z)
    z0 = z - n
    if z0 in lookup:
        return lookup[z0] + Add(*[1/(z0 + k) for k in range(n)])
elif z < 0:
    n = floor(1 - z)
    z0 = z + n
    if z0 in lookup:
        return lookup[z0] - Add(*[1/(z0 - 1 - k) for k in range(n)])
```

### PR DESCRIPTION
**Pull request by CodeMaxx**
Fixes #806
@isuruf @Sumith1896

### CODE REVIEW

**Comment by certik** [comment]
The code looks good to me, but I didn't check the correctness of the math itself. @isuruf does this return correct results?

**Comment by isuruf** [comment]
Looks correct to me. Can you fix `is_canonical` as well?

**Comment by CodeMaxx** [comment]
@isuruf Fixed is_cannonical.

**Comment by CodeMaxx** [comment]
@isuruf Why are we returning `false` when there are valid inputs such as (0,1/4) ? Isn't it canonical? Since I was not very sure I've kept it `false` in the last commit.

**Comment by isuruf** [comment]
A `Polygamma(0, 1/4)` is not a valid input. It should automatically simplify

**Comment by CodeMaxx** [comment]
Oh so it should simplify to the values defined in `polygamma` function( -pi/2 - 3log(2) - EulerGamma).. right?

**Comment by abhinavagarwal07** [comment]
yes.

**Comment by CodeMaxx** [comment]
@isuruf Can you review 'is_cannonical` function

File: symengine/functions.cpp [hunk 1]
```diff
@@ -2994,10 +2994,13 @@ bool PolyGamma::is_canonical(const RCP<const Basic> &n, const RCP<const Basic> &
             auto x_ = rcp_static_cast<const Rational>(x);
             auto num = x_->i.get_num();
             auto den = x_->i.get_den();
-            if (den == 2 and (num == 1 or num == 3)) {
+            if (den == 2 and num == 1) {
                 return false;
             }
-            if (den == 4 and num == 1) {
+            if(den == 3 and (num == 1 or num == 2)) {
```
Comment (isuruf) [inline symengine/functions.cpp]: space after `if`

**Comment by certik** [comment]
It looks good. Any idea why Travis didn't check this?

**Comment by CodeMaxx** [comment]
@certik Are you referring to the space after if ?

**Comment by isuruf** [comment]
2a30b9e passes on travis and dbb11b5 includes a formatting change

**Pull request merged**